IDE/QNX/README.md

```diff
@@ -0,0 +1,147 @@
+# wolfTPM with QNX
+
+Instructions for creating a QNX Momentics wolfTPM Project...
+
+## Create a new QNX Application
+
+1) Create folder for libraries (lib) and includes (inc)
+2) Add library sources into "lib" directory as "wolfssl" and "wolftpm"
+3) Edit Makefile to build sources and include directories.
+
+```
+# wolfSSL and wolfTPM library includes/sources
+INCLUDES += -I./inc -I./lib/wolftpm -I./lib/wolfssl
+CCFLAGS_all += -DWOLFSSL_USER_SETTINGS -DWOLFTPM_USER_SETTINGS
+
+SRCS += $(call wildcard, lib/wolfssl/wolfcrypt/src/*.c)
+SRCS += $(call wildcard, lib/wolfssl/wolfcrypt/src/port/arm/*.c)
+SRCS += $(call wildcard, lib/wolfssl/wolfcrypt/src/port/xilinx/*.c)
+SRCS += $(call wildcard, lib/wolftpm/src/*.c)
+
+# The QNX SPI Driver
+LIBS += -lspi-master
+```
+
+4) Create `inc/user_settings.h` for all wolf specific settings:
+
+Here is a template:
+
+```
+#ifndef WOLF_USER_SETTINGS_H
+#define WOLF_USER_SETTINGS_H
+
+/* TPM */
+#define WOLFTPM_AUTODETECT
+#define TPM_TIMEOUT_TRIES 10000
+#define WOLFTPM_SMALL_STACK
+#if 0
+    #define DEBUG_WOLFTPM
+    #define WOLFTPM_DEBUG_VERBOSE
+    #define WOLFTPM_DEBUG_IO
+    #define WOLFTPM_DEBUG_TIMEOUT
+#endif
+
+/* Platform */
+#define WOLFCRYPT_ONLY
+#define NO_FILESYSTEM
+#define WOLFSSL_IGNORE_FILE_WARN
+
+/* Math */
+#define ECC_TIMING_RESISTANT
+#define TFM_TIMING_RESISTANT
+#define USE_FAST_MATH
+#define FP_MAX_BITS (2 * 4096)
+#define WOLFSSL_NO_HASH_RAW
+#define ALT_ECC_SIZE
+
+/* Enables */
+#define HAVE_ECC
+#define ECC_SHAMIR
+#define HAVE_AESGCM
+#define GCM_TABLE_4BIT
+
+/* Disables */
+#define NO_WOLFSSL_MEMORY
+#define NO_ASN
+#define NO_ASN_TIME
+#define NO_CODING
+#define NO_CERTS
+#define NO_PSK
+
+#define NO_PWDBASED
+#define NO_DSA
+#define NO_RC4
+#define NO_MD4
+#define NO_MD5
+#define NO_SHA
+#define NO_HC128
+#define NO_RABBIT
+#define NO_DES3
+
+#endif /* !WOLF_USER_SETTINGS_H */
+```
+
+5) wolfTPM HAL: Use either `tpm_io.c` directly or copy the required HAL interface into your own .c.
+
+6) Add wolfTPM example code into your own .c.
+
+## QNX SPI Master Patch for Manual Chip Select
```

Comment:
Please add a explanation of why manual chip select is necessary

Reply:
Added